### ISSUE TITLE
async onValueChange called twice 

### ISSUE THREAD

**Issue by AregShahbazian** [lawnstarter/react-native-picker-select#311]
Possibly duplicate of [this](https://github.com/lawnstarter/react-native-picker-select/issues/265) issue.

**Describe the bug**<br />
When `onValueChange` awaits an async call, the delay causes it to be fired twice. Once from the manual select, and once from `getDerivedStateFromProps`

**To Reproduce**<br />

1.  Create a `RNPickerSelect` 
2.  set `onValueChange` to be an async function that awaits something that takes 1000 ms
3.  Try to change the value
4.  Value will not change, or it will flicker back and forth

**Expected behavior**<br />
Only 1 call of `onValueChange`

**Screenshots**<br />
n/a

**Additional details**<br />
-   Device: android
-   OS: Ubuntu
-   react-native-picker-select version: 7.0.0
-   react-native version: 0.62.2
-   react version: 16.11.0

**Reproduction and/or code sample**<br />

Example repo [here](https://github.com/AregShahbazian/RNPickerSelectBug)

**Suggestion for fix**

Awaiting the call [here](https://github.com/lawnstarter/react-native-picker-select/blob/3ee456aa83dd64e52c945bef9c5e67e3aea8d37a/src/index.js#L195) seems to fix the problem. Not sure if this breaks anything else though.

```
    async onValueChange(value, index) {
        const { onValueChange } = this.props;

        await onValueChange(value, index);

        this.setState((prevState) => {
            return {
                selectedItem: prevState.items[index],
            };
        });
    }
```

**Comment by michaelkleinhenz** [comment]
+1 on this. I think I have the same issue. I am updating multiple states in the `onValueChange()` method and this causes the picker to not show the currently selected value. Instead, if "flickers" back to the first entry in `items`.

**Comment by vreuling-bcs** [comment]
Probably related to #112 ?

**Comment by lfkwtz** [comment]
fixed in 8.0.3
